### PR TITLE
Fix/change objectid to int

### PR DESCRIPTION
**Pull request by EricChen-Lei**
# PR Details
## Description

Change objectid type from string to int
## Related Jira Key
CDAR-576

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: co-simulation/fed/mosaic-infrastructure/src/test/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstanceManagerTest.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ public void testOnDetectedObject() throws IOException{
                 0.5,
                 "sensor6",
                 "projection String",
-                "Object1",
+                101,
```
Comment (MishkaMN) [inline co-simulation/fed/mosaic-infrastructure/src/test/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstanceManagerTest.java:152]: Is this correct? Original one is using "Object1" which seems to correlate with 100

File: co-simulation/lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/detector/DetectedObject.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public final class DetectedObject implements Serializable {
      * @param angularVelocity   angular velocity of detected object in sensor/detector frame.
      * @param size              size of object including height,width and length.
      */
-    public DetectedObject(DetectionType type, double confidence, String sensorId, String projString, String objectId,
+    public DetectedObject(DetectionType type, double confidence, String sensorId, String projString, int objectId,
```
Comment (MishkaMN) [inline co-simulation/lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/detector/DetectedObject.java:70]: The line 61 about the **objectId** description need to change. It is saying it is unique string ID

File: co-simulation/lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/detector/DetectedObject.java [hunk 3]
```diff
@@ -216,7 +216,7 @@ public void setProjString(String projString) {
      * Setter for detected object unique string ID.
```
Comment (MishkaMN) [inline co-simulation/lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/detector/DetectedObject.java:216]: int ID

File: co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java [hunk 4]
```diff
@@ -207,7 +207,7 @@ public void testEquals() {
                 0.6,
                 null,
                 null,
-                null,
+                100,
```
Comment (MishkaMN) [inline co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java:210]: is this correct? why is it not null but 100? does it matter?

File: co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java [hunk 5]
```diff
@@ -218,7 +218,7 @@ public void testEquals() {
...
```
Comment (MishkaMN) [inline co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java:221]: is this correct? why is it not null but 100? does it matter?

**Review by MishkaMN — CHANGES_REQUESTED** [review]

**Review by MishkaMN — APPROVED** [review]

File: co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java (hunk 5 above)
Comment (EricChen-Lei) [reply co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java:221]: unit testing passed, shouldn't be a problem

File: co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java (hunk 4 above)
Comment (EricChen-Lei) [reply co-simulation/lib/mosaic-objects/src/test/java/org/eclipse/mosaic/lib/objects/detector/DetectedObjectTest.java:210]: unit testing passed, shouldn't be a problem

**Review by paulbourelly999 — APPROVED** [review]

**Pull request merged**